include/llama.h

```diff
@@ -1197,7 +1197,12 @@ extern "C" {
     LLAMA_API struct llama_sampler * llama_sampler_init_grammar(
             const struct llama_vocab * vocab,
                           const char * grammar_str,
-                          const char * grammar_root);
+                          const char * grammar_root,
+                                  bool lazy,
+                         const char ** trigger_words,
+                                size_t num_trigger_words,
+                   const llama_token * trigger_tokens,
+                                size_t num_trigger_tokens);
```

Comment:
Add a `DEPRECATED` overload of this call to make it easier for developers to adjust to the change.

Reply:
Ah, we break the `llama_sampler_init_...` pattern if we do it like this. Maybe like this instead:

```c
llama_sampler_init_grammar(...); // same as master
llama_sampler_init_grammer_lazy(...); // same as PR but without `bool lazy`?
```

p.s. huh, naming things is difficult 😄

Reply:
Oops, updated, thanks!